### ISSUE TITLE
Whole Foods and Amazon Fresh orders show zero totals

### ISSUE THREAD

**Issue by glenviewjeff** [philipmulcahy/azad#154]
The entries for Whole Foods and Amazon Fresh orders show zeroes for all of the dollar-amount fields, but the data is there in the order history, and even there in the csv file. The payment field for each of these types of orders shows in one case as an example, "Visa ending in xxxx: February 9, 2021: $2.00; Visa ending in xxxx: February 7, 2021: $35.72"

It would be great if you can extract the data for these orders and populate the total if possible.

Thanks for all of your work on this extension!

**Comment by philipmulcahy** [comment]
You’re going to have to help me a bit more to get this one fixed...
1) screenshots 
2) debug dump from an order that manfests the behaviour (view the README.md file for hints on how to do this)

**Comment by glenviewjeff** [comment]
Sorry for the slow response; the email notification got sent to a folder I don't check regularly. I'm not sure exactly what you're looking for screenshots of, but here are a few that may be helpful. Let me know where to send you the json file privately. Thanks!

![image](https://user-images.githubusercontent.com/334914/111874173-f3cc0800-8961-11eb-82c3-93e7291ac39e.png)
![image](https://user-images.githubusercontent.com/334914/111874211-20801f80-8962-11eb-8489-5b7c4e994e7c.png)
![image](https://user-images.githubusercontent.com/334914/111874387-0c88ed80-8963-11eb-95e7-e8652910b53d.png)

**Comment by philipmulcahy** [comment]
OK - that’s useful. I’m also going to need an order dump json file from the order ending in 023. Please send it as an attachment to ***@***.*** ***@***.***> (don’t attach it here), and then post back here to tell me to pick it up - I don’t monitor that address.

Yours,

Philip

**Comment by glenviewjeff** [comment]
Your contact info got bleeped out. Maybe send in a paste bin link or
something?

**Comment by glenviewjeff** [comment]
Found your email address in git logs, please check

**Comment by philipmulcahy** [comment]
Ah-ha!
Bleeping out Nice feature from github.
I first was panicking that I'd doxed myself by leaking my personal email into git - phew! Maybe I still have, but the most prominent email address in recent logs is correct: azadextension at gmail com.
I don't see any email in there from you that isn't via github, and I don't see a json attachment. That's the piece that should enable me to see the html that azad's code is failing to scrape correctly.
Yours,
Philip

**Comment by glenviewjeff** [comment]
Yes I sent to that gmail address with the attachment and got the reply to
tell you to check...

**Comment by philipmulcahy** [comment]
Got it - thanks!
I suspect it might be a few days before I can focus on this - I will report back.

**Comment by glenviewjeff** [comment]
No rush at all!

**Comment by philipmulcahy** [comment]
I have a branch that seems to be getting the key things right, though "key" might be just my perspective.
It seems to get the right order total and tax, and is getting item names, links, quantities and prices.
It is picking the name of the supermarket rather than you as the target.
I probably need to pause for a few days - do you want to play with a test version (that does the above) in the mean time?

**Comment by glenviewjeff** [comment]
Sure, thanks so much! You will have to walk me through how to install etc.
Thanks!

**Comment by philipmulcahy** [comment]
Fetch https://www.mulcahyfamily.org/azad/azad_7916f50b.zip and then take a look at https://github.com/philipmulcahy/azad/blob/master/README.md#how-to-try-out-an-experimental-version

**Comment by glenviewjeff** [comment]
Thanks much.. just installed but getting the same behavior as before. I did
disable the previous extension. When I first ran the new extension, my most
recent Whole Foods order didn't even show up in the list at all.  I cleared
the cache after running it the first time, and it showed up, but in the
same fashion as the release version.

[image: image.png]

**Comment by philipmulcahy** [comment]
That is strange - I have a solid simulated scrape for 112-1097135-4205023 based on your json file.

Would you mind trying this one? https://www.mulcahyfamily.org/azad/azad_719e849f.zip

**Comment by philipmulcahy** [comment]
If it doesn't work, would you mind sharing a console log for a clean (empty cache) run?

**Comment by glenviewjeff** [comment]
Sure, but I wonder if this is helpful first? I just noticed there are a bunch of errors on the extensions page for this version. 

![image](https://user-images.githubusercontent.com/334914/112769144-253d7700-8fe5-11eb-9f7f-8cf7e941eb6c.png)

**Comment by philipmulcahy** [comment]
without context, they are hard to interpret

**Comment by glenviewjeff** [comment]
OK, just sent to your az email, so please check. Thanks!

**Comment by philipmulcahy** [comment]
qq: what happens when you visit these two links?

https://www.amazon.com/uff/your-account/order-details/ref=ppx_yo_dt_b_uff_or_o07?_encoding=UTF8&orderID=112-1097135-4205023

https://www.amazon.com/uff/your-account/order-details/ref=ppx_yo_dt_b_uff_or_o02?_encoding=UTF8&orderID=112-1097135-4205023

**Comment by glenviewjeff** [comment]
They both show the same order. Do me a favor and edit them out/delete the
reply; I don't want my order #'s out on the web for security reasons.
Thanks!

**Comment by philipmulcahy** [comment]
Done - order IDs are pretty useless on their own btw.

**Comment by ScottMcNay** [comment]
Order IDs appear to be issued randomly, aside from starting with "1" for
physical orders and "D" for digital orders; they only have meaning to you.

--Scott.

**Comment by glenviewjeff** [comment]
Just want to make sure you're not waiting on something from me. Thanks!

**Comment by philipmulcahy** [comment]
No. I need to find quality time with my laptop. I’m guessing the next step might be another special build with logging to help identify the problem. 
I was so confident I’d got it working with that last version.
Thank you for being proactive!

Yours,

Philip

**Comment by philipmulcahy** [comment]
OK - I've got an hour or two tonight, and I've taken a better look at the log you provided on the 28th so that I can figure out what extra logging statements I need to add. Unfortunately the log is incomplete, and doesn't contain the order you provided me as a dump file. I think this may have happened because you didn't open the console until after the scrape had started - chrome (understandably, so as to avoid using memory unnecessarily) only retains the most recent few lines if you don't have the console open. Would you mind taking another shot at capturing the log with azad_719e849f.zip ?

**Comment by glenviewjeff** [comment]
Sure, will do now. Sorry I only now saw your email if the hour or two
already elapsed. :)

**Comment by glenviewjeff** [comment]
Check your email, thanks!

**Comment by glenviewjeff** [comment]
Check your email again.. ran the new debug version. Thanks!

**Comment by glenviewjeff** [comment]
Emailed you back but in case you didn't check.. "Wait, but I just looked at the output (before clicking show items not orders) and I'm seeing that it now appears to be capturing all of the whole foods orders. It does also list each item when I check that option. Did I make a mistake on the prior version you'd sent, and maybe it worked fine? I don't recall what the situation was last time. Sorry if I led you astray!!"

**Comment by philipmulcahy** [comment]
I'm gonna let this ticket sit for a bit, on the basis that 0a7d44af has what we need, and it's on the master branch. I'll close when I push a release to the chrome web store (unless @glenviewjeff tells me otherwise).

**Comment by glenviewjeff** [comment]
Emailed: "I'm very confused; didn't you tell me that you only added instrumenting and nothing else between the two versions you sent me? The checkbox option isn't there, and Whole Foods orders aren't working in any other versions I have. 

I reverted back to the prior version just now (ID: alacaeojcbgphedaobkbeadnmgafggfk ???), and it appears to still be broken, cleared cache, still showing zeroes for whole foods. In the future, though pedantic, might help to give a minor revision to development versions to distinguish them. Also verified that it doesn't work in the other version I have installed, which is likely the release version (ID: mgkilgclilajckgnedgjgnfdokkgnibi). 

By the way, that's a pretty cool feature to have a checkbox for item vs. order, so please keep it if possible. And thanks for all of your help; I made a small donation to the charity you suggested. "

**Comment by philipmulcahy** [comment]
I was trying to fix another bug (another ticket) at the same time, and the items work came in as a side effect, because I didn’t want to let the three branches diverge too much. It is a bit strange that your issue got fixed in the version it did ‘cos I’ve been passing the test with your grocery order for all of the ones I’ve sent you.

**Comment by philipmulcahy** [comment]
I've built v1.9.21 and submitted it for review by the chrome web store - recent releases have taken 12 to 24 hours to make it through to the store.

**Comment by glenviewjeff** [comment]
That's really weird.. I double-checked that it didn't work with the prior test version. Whatever the case, hopefully it will work in the release version. Thanks for all your help!

**Comment by glenviewjeff** [comment]
Updated Chrome store version, worked beautifully, thanks!
